### PR TITLE
doc: zigbee|matter: add section about memory footprint

### PR DESCRIPTION
**Pull request by greg-fer**
Added Zigbee section to Memory footprint optimization page.
KRKNWK-11442.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/app_memory.rst [hunk 1]
```diff
@@ -96,3 +97,19 @@ Complete the following actions to optimize the Thread part of your application:
...
+
+You can complete the following actions to optimize the :ref:`Zigbee <ug_zigbee>` part of your application:
+
+* Disable features that your application does not use.
+  For example, disable the following features:
+
+  * Asserts
+  * Logging
+
+* Make sure the :kconfig:option:`CONFIG_RAM_POWER_DOWN_LIBRARY` is set to ``y``.
```
Comment (tomchy) [inline doc/nrf/app_memory.rst:114]: This feature optimizes the RAM power usage, not the RAM footprint

**Review by sebastiandraus — APPROVED** [review]

File: doc/nrf/app_memory.rst [hunk 2]
```diff
@@ -62,19 +68,28 @@ Complete the following actions to optimize the Bluetooth® part of your applicat
...
 
-Complete the following actions to optimize the Thread part of your application:
+Besides applying `General recommendations`_, you can also complete the following actions to optimize the :ref:`Matter <ug_matter>` part of your application:
 
-* Disable features that your application does not use.
-  For example, disable the following features:
+* Make sure Zephyr's :ref:`zephyr:shell_api` are disabled for your application.
+  These are listed in a dedicated section in each Matter sample's :file:`prj.conf` file.
+* Check `Thread`_ memory footprint optimization actions, as the Matter application layer uses OpenThread.
+
```
Comment (kkasperczyk-no) [inline doc/nrf/app_memory.rst:81]: Some other optimizations that can be made:

- Change default log level of Zephyr modules from info to warning (CONFIG_LOG_DEFAULT_LEVEL=2). That can be done assuming user logs doesn't use default log level, but define own one.
-  Change log level of Matter logs from debug to info (CONFIG_MATTER_LOG_LEVEL_INFO=y)
- Change verbosity of assert messages to lower value (CONFIG_ASSERT_VERBOSE=n)

File: doc/nrf/app_memory.rst [hunk 3]
```diff
@@ -62,19 +68,28 @@ Complete the following actions to optimize the Bluetooth® part of your applicat
...
-Thread
+Matter
 ******
 
-Complete the following actions to optimize the Thread part of your application:
+Besides applying `General recommendations`_, you can also complete the following actions to optimize the :ref:`Matter <ug_matter>` part of your application:
 
-* Disable features that your application does not use.
-  For example, disable the following features:
+* Make sure Zephyr's :ref:`zephyr:shell_api` are disabled for your application.
```
Comment (markaj-nordic) [inline doc/nrf/app_memory.rst:78]:
```suggestion
* Make sure Zephyr's :ref:`zephyr:shell_api` is disabled for your application.
```

File: doc/nrf/app_memory.rst [hunk 4]
```diff
@@ -62,19 +68,28 @@ Complete the following actions to optimize the Bluetooth® part of your applicat
...
+  These are listed in a dedicated section in each Matter sample's :file:`prj.conf` file.
```
Comment (markaj-nordic) [inline doc/nrf/app_memory.rst:79]:
```suggestion
  Related configuration options are listed in a dedicated section in each Matter sample's :file:`prj.conf` file.
```

Comment (markaj-nordic) [inline doc/nrf/app_memory.rst:79]: Maybe it's worth adding that for each sample there is a [prj_release.conf](https://github.com/nrfconnect/sdk-nrf/blob/main/samples/matter/light_bulb/configuration/nrf52840dk_nrf52840/prj_release.conf) with can be used as a first aid to reduce the footprint.

**Comment by greg-fer** [comment]
@kkasperczyk-no , @markaj-nordic , please recheck.

**Review by markaj-nordic — APPROVED** [review]

**Pull request merged**